plugins/SpectrumAnalyzer/SaSpectrumView.cpp

```diff
@@ -348,10 +351,28 @@ QPainterPath SaSpectrumView::makePath(std::vector<float> &displayBuffer, float r
 	// Bins falling to interval [x_start, x_next) contribute to a single point.
 	float max = m_displayBottom;
 	float x_start = -1;		// lower bound of currently constructed point
+
+	// Speed up bin → x position translation by building a LUT cache.
+	// Update the cache only when range or display width are changed.
+	float rangeMin = m_processor->getFreqRangeMin(m_controls->m_logXModel.value());
+	float rangeMax = m_processor->getFreqRangeMax();
+	if (rangeMin != m_cachedRangeMin || rangeMax != m_cachedRangeMax || m_displayWidth != m_cachedDisplayWidth)
+	{
+		m_cachedBinToX.resize(m_processor->binCount() + 1);
+		m_cachedBinToX.clear();
```

Comment:
Clearing the vector before overwriting everything looks redundant. Also, isn't it possible to reserve the memory space on construction? it's only about 32k per an instance.

Reply:
Ah, you are right, I did not even realize the clear is completely pointless.

As for the resize itself: first I was worried that the constant could be overlooked when someone decides to increase the maximum FFT block size a decade later (since everything else follows the `processor->binCount()`). But I guess I can simply check the last element of `FFT_BLOCK_SIZES` and allocate the LUT based on that. It could slightly improve performance while the window is being resized (since each pixel-sized step of the resize triggers cache update), so why not. I wonder if it even makes sense to use `array` instead of `vector`, although I doubt it will bring any performance benefit.

In other news, your comment made me realize that there is one more variable that can invalidate the cache: the `binCount`. When the number of bins changes, the mapping of bin to frequency also changes, so the LUT must be rebuilt.

Reply:
Updated.
I kept the cache a `vector`, since the size of `array` could not be defined by the last element from the list of allowed FFT block sizes (the compiler does not consider it constant enough at compile time).